frontend/src/metabase/visualizations/components/Visualization.jsx

```diff
@@ -224,20 +222,22 @@ export default class Visualization extends Component<*, Props, State> {
     }
 
     handleOnChangeCardAndRun = (card: UnsavedCard) => {
-        // If the current card is saved, carry that information to the new card for showing lineage
-        const { series } = this.state
+        const { series, clicked } = this.state;
+
+        // If the current card is saved or is based on a saved question,
+        // carry that information to the new card for showing lineage
+        const index = (clicked && clicked.seriesIndex) || 0;
         // $FlowFixMe
-        const currentlyInSavedCard = series[0] && series[0].card && series[0].card.id
-        if (currentlyInSavedCard) {
-            const savedCard: CardObject = {
+        const hasOriginalCard = series[index] && series[index].card && (series[index].card.id || series[index].card.original_card_id);
```

Comment:
Is there a nicer way to handle these cascading checks? We seem to do a lot of them throughout the code base where we're checking multiple levels of existence and it makes things a bit gnarly to read.